### PR TITLE
minor sql touchups

### PR DESCRIPTION
**Pull request by tbg**
- set context to insecure on 'http' url
- fix select when result set empty

we should start running tests against an actual KV. Do we just start a TestServer as usual?

### CODE REVIEW

**Comment by petermattis** [comment]
Can you add a test case to `sql/driver/driver_test.go? That's where we start up a server that we can run SQL statements against.

**Comment by petermattis** [comment]
LGTM

File: sql/driver/driver.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ func (d *roachDriver) Open(dsn string) (driver.Conn, error) {
 	}
 	ctx := &base.Context{}
 	ctx.InitDefaults()
+	ctx.Insecure = u.Scheme == "http"
```
Comment (MBerhault) [inline sql/driver/driver.go]: This shouldn't be needed, there are two separate registered senders: one for http, the other for https. They each specify a function that sets the Insecure flag from the scheme. See http_sender.go:Init()
This is a verbatim copy of the client/sender functionality. If the context doesn't get properly set, we should figure out why.

Comment (tschottdorf) [inline sql/driver/driver.go]: Let me check again to see what goes wrong, if anything.

Comment (MBerhault) [inline sql/driver/driver.go]: nevermind, found it. L52 in this file should be calling newSender, not newHTTPSender. It bypasses the sender registration and does not set the context appropriately.

**Comment by MBerhault** [comment]
LGTM. maybe add a TODO for a test of insecure mode.

**Comment by petermattis** [comment]
LGTM, though you accidentally added the `sql_bank` binary in one of the commits.

**Comment by tschottdorf** [comment]
whoops. Fixed.

**Pull request merged**